recipes_source/recipes/tuning_guide.py

```diff
@@ -0,0 +1,376 @@
+"""
+Performance Tuning Guide
+*************************
+**Author**: `Szymon Migacz <https://github.com/szmigacz>`_
+
+Performance Tuning Guide is a set of optimizations and best practices which can
+accelerate training and inference of deep learning models in PyTorch. Presented
+techniques often can be implemented by changing only a few lines of code and can
+be applied to a wide range of deep learning models across all domains.
+
+General optimizations
+---------------------
+"""
+
+###############################################################################
+# Enable async data loading and augmentation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# `torch.utils.data.DataLoader <https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader>`_
+# supports asynchronous data loading and data augmentation in separate worker
+# subprocesses. The default setting for ``DataLoader`` is ``num_workers=0``,
+# which means that the data loading is synchronous and done in the main process.
+# As a result the main training process has to wait for the data to be available
+# to continue the execution.
+#
+# Setting ``num_workers > 0`` enables asynchronous data loading and overlap
+# between the training and data loading. ``num_workers`` should be tuned
+# depending on the workload, CPU, accelerator, and location of training data.
+#
+# ``DataLoader`` accepts ``pin_memory`` argument, which defaults to ``False``.
+# When using an accelerator it's better to set ``pin_memory=True``, this
```

Comment:
I think you can safely specify `accelerator` to `GPU` here. TPU via pytorch-xla works quite differently: the training script executes in a GCP VM on one machine, and the TPUs are attached in another machine (VM sends the input data and the IR to TPUs' machine via Ethernet). I don't think setting `pin_memor=True` for TPU do much (cuz this would refers to the VM), and pytorch-xla has its own dataloader to pipeline data uploading and graph execution.

Reply:
makes sense, I'll switch it back to `GPU`.